compiler/plugins/target/ROCM/test/opt_pass_plugin/GPUHello.cpp

```diff
@@ -0,0 +1,78 @@
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/Pass.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace {
+
+struct GpuHello final : llvm::PassInfoMixin<GpuHello> {
+  llvm::PreservedAnalyses run(llvm::Module &module,
+                              llvm::ModuleAnalysisManager &) {
+    bool modifiedCodeGen = runOnModule(module);
+    if (!modifiedCodeGen)
+      return llvm::PreservedAnalyses::all();
+
+    return llvm::PreservedAnalyses::all();
```

Comment:
The two outcomes are the same here and something doesn't add up -- either the `if` is redundant or the return values are wrong.

Reply:
This was originally a ternary operation that seems to have lost something in translation:
```
    return (modifiedCodeGen ? llvm::PreservedAnalyses::none()
                            : llvm::PreservedAnalyses::all());
``` 
```suggestion
if (modifiedCodeGen)
     llvm::PreservedAnalyses::none();
else
    llvm::PreservedAnalyses::all();
```

Reply:
Damn, sorry